### PR TITLE
Documentation Update

### PR DESCRIPTION
**Pull request by csawye01**
Many thanks for contributing to nf-core/nanoseq!

Please fill in the appropriate checklist below (delete whatever is not relevant). These are the most common things requested on pull requests (PRs).

## PR checklist
 - [X] This comment contains a description of changes (with reason)
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] If necessary, also make a PR on the [nf-core/nanoseq branch on the nf-core/test-datasets repo]( https://github.com/nf-core/test-datasets/pull/new/nf-core/nanoseq)
 - [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
 - [ ] Make sure your code lints (`nf-core lint .`).
 - [X] Documentation in `docs` is updated
 - [ ] `CHANGELOG.md` is updated
 - [X] `README.md` is updated

**Learn more about contributing:** https://github.com/nf-core/nanoseq/tree/master/.github/CONTRIBUTING.md

### CODE REVIEW

**Comment by MaxUlysse** [comment]
Shouldn't this PR be made on the `dev` branch instead?

File: docs/output.md [hunk 1]
```diff
@@ -22,32 +23,46 @@ The pipeline is built using [Nextflow](https://www.nextflow.io/) and processes d
...
   FastQ files output for each barcode
-* `guppy/unclassified/`  
+* `guppy/basecalling/unclassified/`  
   FastQ files output that are unclassified
-* `guppy/sequencing_summary.txt`
+* `guppy/fastq`
+  Merged output of fastq files into one fastq for each barcode
 
-## Quality Control
+## Sequncing Quality Control
```
Comment (Zethson) [inline docs/output.md]: Typo?
-> Sequencing

Comment (MaxUlysse) [inline docs/output.md]:
```suggestion
## Sequencing Quality Control
```

File: docs/output.md [hunk 2]
```diff
@@ -22,32 +23,46 @@ The pipeline is built using [Nextflow](https://www.nextflow.io/) and processes d
 [Guppy](https://nanoporetech.com/nanopore-sequencing-data-analysis)
 
 *Description*:  
-Guppy will demultiplex and barcode the data given from an ONT device. The flowcell, kit and barcode kit must be given in the command line if demultiplexing needed. This step can by bypassed using the `--skip_demultiplexing` parameter when initiating the pipeline. The output folders will be separated into the barcodes from the kit used and unclassified.
+Guppy will demultiplex and barcode the data given from an ONT device. The flowcell, kit and barcode kit must be given in the command line if demultiplexing needed. This step can by bypassed using the `--skip_demultiplexing` parameter when initiating the pipeline. The output folders will be separated into the barcodes from the kit used and unclassified. The output in each barcode folder is then merged into one fastq file for easier downstream processesing.
```
Comment (MaxUlysse) [inline docs/output.md]:
```suggestion
Guppy will demultiplex and barcode the data given from an ONT device. The flowcell, kit and barcode kit must be given in the command line if demultiplexing needed. This step can by bypassed using the `--skip_demultiplexing` parameter when initiating the pipeline. The output folders will be separated into the barcodes from the kit used and unclassified. The output in each barcode folder is then merged into one fastq file for easier downstream processing.
```

File: docs/output.md [hunk 3]
```diff
@@ -14,6 +14,7 @@ The pipeline is built using [Nextflow](https://www.nextflow.io/) and processes d
 * [GraphMap](#graphmap) - mapping for long reads
 * [MiniMap2](#minimap2) - mapping for long reads
 * [SortBam](#sortbam) - coordinate sort BAM files using SAMtools
+* [bedtools](#bedtools) - Create bigWig and bigBed files
```
Comment (MaxUlysse) [inline docs/output.md]:
```suggestion
* [bedtools](#bedtools) - create bigWig and bigBed files 
```

File: docs/output.md [hunk 4]
```diff
@@ -22,32 +23,46 @@ The pipeline is built using [Nextflow](https://www.nextflow.io/) and processes d
...
+PycoQC and NanoPlot give general quality metrics about the sequencing run. It provides information about the distribution of read length, read length over time, number of reads per barcode and other general stats.
+![PycoQC - Number of Reads per Barcode plot](images/NumberofReadsperBarcode.png)
 
 *Output directories*:
 
 * `pycoQC/`  
   An .html file output is produced that includes a run summary and graphical representation of distribution of read length, distribution of read quality scores, mean read quality per sequence length, output per channel over experiment time, output over experiment time, read quality over experiment time, readlength over experiment time, and percentage of reads per barcode.
 * `nanoplot/summary/`  
-  An output of a .png file of a statistical summary and an html summary file.
+  An output of a .png file of a statistical run summary and an html summary file of overall run.
```
Comment (MaxUlysse) [inline docs/output.md]: What do you mean by `statistical run summary` and `overall run`?

File: docs/output.md [hunk 5]
```diff
@@ -22,32 +23,46 @@ The pipeline is built using [Nextflow](https://www.nextflow.io/) and processes d
...
+
```
Comment (MaxUlysse) [inline docs/output.md]:
```suggestion
```

File: docs/output.md [hunk 6]
```diff
@@ -59,15 +74,31 @@ The FastQ reads are mapped to the given reference assembly provided using either
...
 * `<ALIGNER>/samtools_stats/`  
   `*.flagstat`, `*.idxstats` and `*.stats` files generated from the alignment files using SAMtools.
 
+## bigWig and bigBed 
+
+*Documentation*:
+[`BEDTools`](https://github.com/arq5x/bedtools2/), [`bedGraphToBigWig`](http://hgdownload.soe.ucsc.edu/admin/exe/), [`bedToBigBed`](http://hgdownload.soe.ucsc.edu/admin/exe/)
+
+*Description*:
+Creation of bigWig and bigBed coverage tracks for visualisation. This can by bypassed by setting the parameters `--skip_bigwig` and/or `--skip_bigbed`.
```
Comment (MaxUlysse) [inline docs/output.md]:
```suggestion
Creation of bigWig and bigBed coverage tracks for visualisation. This can be bypassed by setting the parameters `--skip_bigwig` and/or `--skip_bigbed`.
```

File: docs/output.md (hunk 4 above)
Comment (csawye01) [reply docs/output.md]: Just to differentiate it between the individual fastqc Nanoplot report from the overall Nanoplot report. I will reword this to be more clear.

Comment (MaxUlysse) [reply docs/output.md]: That would be perfect, because I honestly had no idea what you were talking about.
:confused:

**Comment by MaxUlysse** [comment]
I think you just need to resolve the merge conflicts on `docs/output.md` and you should be good to go.

File: docs/usage.md [hunk 7]
```diff
@@ -269,7 +276,6 @@ The following options make this easy:
 | `--skip_qc`             | Skip all QC steps apart from MultiQC |
 | `--skip_pycoqc`         | Skip pycoQC                          |
 | `--skip_nanoplot`       | Skip NanoPlot                        |
-| `--skip_fastqc`         | Skip FastQC                          |
```
Comment (drpatelh) [inline docs/usage.md]: Let's leave this is in for now until we decide to remove it everywhere.

File: docs/usage.md [hunk 8]
```diff
@@ -31,7 +31,13 @@
   * [`--save_align_intermeds`](#--save_align_intermeds)
   * [`--skip_alignment`](#--skip_alignment)
 * [Coverage tracks](#coverage-tracks)
+  * [`--skip_bigwig`](#--skip-bigwig)
```
Comment (drpatelh) [inline docs/usage.md]: These were removed because they have been put in a table and so won't have explicit bullets in the relevant sections so the links won't work.

File: docs/usage.md [hunk 9]
```diff
@@ -31,7 +31,13 @@
...
+  * [`--skip_bigbed`](#--skip-bigbed)
 * [Skipping QC steps](#skipping-qc-steps)
+  * [`--skip_qc`](#--skip-qc)
```
Comment (drpatelh) [inline docs/usage.md]: Same as above comment.

File: docs/usage.md [hunk 10]
```diff
@@ -78,6 +84,7 @@ nextflow run nf-core/nanoseq \
     --flowcell FLO-MIN106 \
     --kit SQK-LSK109 \
     --barcode_kit SQK-PBK004 \
+    --aligner minimap2 \
```
Comment (drpatelh) [inline docs/usage.md]: This will be the default anyway so best to just leave it out so we can have a minimal usage command.

File: docs/output.md [hunk 11]
```diff
@@ -78,7 +107,9 @@ The pipeline has special steps which allow the software versions used to be repo
 
 * `multiqc/Project_multiqc_report.html`  
   MultiQC report - a standalone HTML file that can be viewed in your web browser
-* `multiqc/Project_multiqc_data/`  
+* `multiqc/multiqc_data/`  
   Directory containing parsed statistics from the different tools used in the pipeline
+* `multiqc/multiqc_plots/`
+  Directory containing the image files of the graphs included in MultiQC
```
Comment (drpatelh) [inline docs/output.md]: We should also include a description of the `pipeline_info` files generated by Nextflow. See https://github.com/nf-core/chipseq/blob/master/docs/output.md#other-results

File: docs/output.md [hunk 12]
```diff
@@ -22,32 +23,45 @@ The pipeline is built using [Nextflow](https://www.nextflow.io/) and processes d
...
 *Output directories*:
 
-* `guppy/barcode*/`  
+* `guppy/basecalling/barcode*/`  
   FastQ files output for each barcode
-* `guppy/unclassified/`  
+* `guppy/basecalling/unclassified/`  
   FastQ files output that are unclassified
-* `guppy/sequencing_summary.txt`
+* `guppy/fastq`
```
Comment (drpatelh) [inline docs/output.md]:
```suggestion
* `guppy/fastq/`
```

File: docs/output.md [hunk 13]
```diff
@@ -59,15 +73,30 @@ The FastQ reads are mapped to the given reference assembly provided using either
...
+## bigWig and bigBed 
+
+*Documentation*:
+[`BEDTools`](https://github.com/arq5x/bedtools2/), [`bedGraphToBigWig`](http://hgdownload.soe.ucsc.edu/admin/exe/), [`bedToBigBed`](http://hgdownload.soe.ucsc.edu/admin/exe/)
+
+*Description*:
+Creation of bigWig and bigBed coverage tracks for visualisation. This can be bypassed by setting the parameters `--skip_bigwig` and/or `--skip_bigbed`.
+
+*Output directories*:
+* `<ALIGNER>/bigwig`
```
Comment (drpatelh) [inline docs/output.md]:
```suggestion
* `<ALIGNER>/bigwig/`  
```

File: docs/output.md [hunk 14]
```diff
@@ -59,15 +73,30 @@ The FastQ reads are mapped to the given reference assembly provided using either
...
+*Documentation*:
+[`BEDTools`](https://github.com/arq5x/bedtools2/), [`bedGraphToBigWig`](http://hgdownload.soe.ucsc.edu/admin/exe/), [`bedToBigBed`](http://hgdownload.soe.ucsc.edu/admin/exe/)
+
+*Description*:
+Creation of bigWig and bigBed coverage tracks for visualisation. This can be bypassed by setting the parameters `--skip_bigwig` and/or `--skip_bigbed`.
+
+*Output directories*:
+* `<ALIGNER>/bigwig`  
+  The bigWig files will be output here.
+* `<ALIGNER>/bigbed`
```
Comment (drpatelh) [inline docs/output.md]:
```suggestion
* `<ALIGNER>/bigbed/`  
```

File: docs/output.md [hunk 15]
```diff
@@ -78,7 +107,27 @@ The pipeline has special steps which allow the software versions used to be repo
...
+  [Nextflow!](https://www.nextflow.io/docs/latest/tracing.html)
+
+  *Description*:  
+  Nextflow provides excellent functionality for generating various reports relevant to the running and execution of the pipeline. This will allow you to trouble-shoot errors with the running of the pipeline, and also provide you with other information such as launch commands, run times and resource usage.
+
+  *Output directories*:
+  * `pipeline_info/`  
+    * Reports generated by the pipeline - `pipeline_report.html`, `pipeline_report.txt` and `software_versions.csv`.
+    * Reports generated by Nextflow - `execution_report.html`, `execution_timeline.html`, `execution_trace.txt` and `pipeline_dag.svg`.
+    * Reformatted design files used as input to the pipeline - `design_reads.csv` and `design_controls.csv`.
```
Comment (drpatelh) [inline docs/output.md]:
```suggestion
    * Reformatted samplesheet files used as input to the pipeline - `samplesheet_reformat.csv`.
```

File: docs/usage.md [hunk 16]
```diff
@@ -78,6 +78,7 @@ nextflow run nf-core/nanoseq \
...
```
Comment (drpatelh) [inline docs/usage.md]:
```suggestion
```

File: docs/output.md [hunk 17]
```diff
@@ -59,15 +73,30 @@ The FastQ reads are mapped to the given reference assembly provided using either
...
 
-*Output directories*:
+![ALIGNER - Alignment per barcode](images/mqc_samtools_alignment_plot_1.png)
 
-* `graphmap2/`  
-  If the `--aligner graphmap2` parameter is used, the results will be output here.
-* `results/minimap2`  
-  If the `--aligner minimap2` parameter is used, the results will be output here.
+*Output directories*:
+* `graphmap/`
```
Comment (drpatelh) [inline docs/output.md]:
```suggestion
* `graphmap2/`  
```

File: docs/output.md [hunk 18]
```diff
@@ -59,15 +73,30 @@ The FastQ reads are mapped to the given reference assembly provided using either
...
-*Output directories*:
+![ALIGNER - Alignment per barcode](images/mqc_samtools_alignment_plot_1.png)
 
-* `graphmap2/`  
-  If the `--aligner graphmap2` parameter is used, the results will be output here.
-* `results/minimap2`  
-  If the `--aligner minimap2` parameter is used, the results will be output here.
+*Output directories*:
+* `graphmap/`  
+  If the `--aligner graphmap` parameter is used, the sorted and indexed bam files will be output here.
```
Comment (drpatelh) [inline docs/output.md]:
```suggestion
  If the `--aligner graphmap2` parameter is used, the sorted and indexed bam files will be output here.
```

**Comment by drpatelh** [comment]
@csawye01 Can we add the `TODO` statements back in until we release please?

**Comment by drpatelh** [comment]
Almost there @csawye01  👍 
https://travis-ci.com/nf-core/nanoseq/jobs/259561385

Couple more `markdownlint` errors to fix.

```
/home/travis/build/nf-core/nanoseq/docs/output.md:132 MD006/ul-start-left Consider starting bulleted lists at the beginning of the line [Expected: 0; Actual: 2]
/home/travis/build/nf-core/nanoseq/docs/output.md:138 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 4]
```

**Comment by drpatelh** [comment]
Lovely jubbly :+1: Tests passing now.
